src/docs/user/ProActiveUserGuide.adoc

```diff
@@ -2611,6 +2611,103 @@ Task Synchronization API can be used for multiple dynamic synchronization patter
 
 A complete example of Producer / Consumer, _ServiceDeployerConsumer_ can be downloaded link:examples/ServiceDeployerConsumer.xml[here].
 
+[[_task_signal_api]]
+=== Signal API
+
+The *Signal API* is a high-level API that enables users to interact with running jobs via signals (we also call them actions).
+That is, it allows a job to expose specific signals (e.g., `soft_stop` to properly finish a job) that can be triggered by users from the *Workflow Execution portal*.
+
+At a high-level view, the Signal API allows jobs and users to exchange signals using a communication channel provided by the <<_task_synchronization_api,synchronization API>>.
+The typical usage scenario of the Signal API is the following:
+
+  . A job declares to be ready to handle one or more signals. The signals are then displayed in the Workflow Execution portal (e.g., display the `soft_stop` signal)
+
+  . The job waits to receive these signals, or check for their reception regularly,
+
+  . A user clicks on the signal displayed in the Workflow Execution portal (e.g., click on the `soft_stop` signal), which sends this signal to the job,
+
+  . At the signal reception, the job executes the corresponding behavior (defined in the workflow).
+
+As already mentioned, the Signal API uses the <<_task_synchronization_api,synchronization API>> to provide a *persistent key-value channel* to store job signals.
+It uses a single channel (called *PA_SIGNALS_CHANNEL*) to store the signals of all jobs. The signal channel contains an entry per job.
+The key of each entry is the job id, whereas the value of the entry is the set of signals used by the considered job.
+Job signals are organized in a link:https://docs.oracle.com/javase/8/docs/api/java/util/Set.html[Java Set] structure,
+which means that a signal appears only once in the set, even if it is sent many times.
+All job signals are automatically removed when the job execution finishes (i.e., the job is in the status Finished, Cancelled, Failed or Killed).
```

Comment:
I think there is some useful and some unecessary information in this section. To make it more clear, you could express implementation details as an unordered list such as:

Signal API implementation makes use of the Synchronization API with the following mechanisms:
 * It uses a single channel (called *PA_SIGNALS_CHANNEL*) to store the signals of all jobs. The signal channel contains one entry per job.
 * ...

Reply:
Changed as requested